src/cascadia/PublicTerminalCore/HwndTerminal.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+
+extern "C" {
+    __declspec(dllexport) HRESULT _stdcall CreateTerminal(HWND parentHwnd, _Out_ void** hwnd, _Out_ void** terminal);
+    __declspec(dllexport) void _stdcall SendTerminalOutput(void* terminal, LPCWSTR data);
+    __declspec(dllexport) void _stdcall RegisterScrollCallback(void* terminal, void __stdcall callback(int, int, int));
+    __declspec(dllexport) HRESULT _stdcall TriggerResize(void* terminal, double width, double height, _Out_ int* charColumns, _Out_ int* charRows);
+    __declspec(dllexport) void _stdcall DpiChanged(void* terminal, int newDpi);
+    __declspec(dllexport) void _stdcall UserScroll(void* terminal, int viewTop);
+    __declspec(dllexport) void _stdcall StartSelection(void* terminal, COORD cursorPosition, bool altPressed);
+    __declspec(dllexport) void _stdcall MoveSelection(void* terminal, COORD cursorPosition);
+    __declspec(dllexport) void _stdcall ClearSelection(void* terminal);
+    __declspec(dllexport) const wchar_t* _stdcall GetSelection(void* terminal);
+    __declspec(dllexport) bool _stdcall IsSelectionActive(void* terminal);
+    __declspec(dllexport) void _stdcall DestroyTerminal(void* terminal);
+
+    enum CursorStyle
+    {
+
+    };
+    struct TerminalSettings
+    {
+        COLORREF DefaultBackground;
+        COLORREF DefaultForeground;
+        CursorStyle CursorShape;
+    };
+    }
+
+struct HwndTerminal {
+public:
+    HwndTerminal(HWND hwnd);
+    ~HwndTerminal();
+    HRESULT Initialize();
+    void SendOutput(std::wstring_view data);
+    HRESULT Refresh(double width, double height, _Out_ int* charColumns, _Out_ int* charRows);

```

Comment:
We're generally walking away from out parameters as pointers and trying to use out refs.

Also, since it's width/height pairs, maybe it should be in a SIZE/COORD type structure?

Reply:
I'd still sort of prefer a SIZE/COORD here. Is there a reason why that's difficult?

Reply:
+1 to the SIZE/COORD

Reply:
I'll get the integer dimensions updates to SIZE/COORD, is there an equivalent that holds `double` values?